ft232/mpsse.py

```diff
@@ -0,0 +1,555 @@
+        This function implements chapter 3.7 - Loopback Commands.
+        It does not return anything but directly writes the Commandbyte to the device.
+        """
+        self.__Ft232.write(bytes([0x84]))
+
+    def disableLoopback(self):
+        """ This disables Loopback. See enableLoopback() for mor details
+        This function implements chapter 3.7 - Loopback Commands.
+        It does not return anything but directly writes the Commandbyte to the device.
+        """
+        self.__Ft232.write(bytes([0x85]))
+
+    def __setClockDivisor(self, Divisor):
+        """ The Clock of the MPSSE device can be changed by setting a divisor, which lowers the Baseclock of the device by following equation:
+        Clock = Baseclock / (( 1 + Divisor ) * 2)
+        This should not be called directly. Instead use _setClock().
+        This function implements chapter 3.8 - Clock Divisor.
+        It does not return anything but directly writes the Commandbyte and new Divisor to the device.
+        """
+        DivisorLowByte = Divisor & 0xff
+        DivisorHighByte = (Divisor >> 8) & 0xff
+        print("clock divisor set to " + str(Divisor))
+        self.__Ft232.write(bytes([0x86, DivisorLowByte, DivisorHighByte]))
+
+    def _cpuMode(self): #feel free to change the name
+        """ To be implemented.
+        See chapter 4 - Instructions for CPU mode for more details
+        """
+        pass
+
+    def _mcuHost(self): #feel free to change the name
+        """ To be implemented.
+        See chapter 5 - Instructions for use in both MPSSE and MCU Host Emulation Modes for more details
+        """
+        pass
+
+    def __enableClockDivide(self):
+        """ This will turn on the Baseclock devide of the MPSSE unit by 5.
+        This function implements chapter 6.2. Enable Clk Divide by 5.
+        It does not return anything but directly writes the Commandbyte and new Divisor to the device.
+        """
+        if self.__isModernDevice():
+            self.__Ft232.write(bytes([0x8b]))
+
+    def __disableClockDivide(self):
+        """ This turn off the Baseclock devide of the MPSSE unit by 5.
+        This function implements chapter 6.1. Disable Clk Divide by 5.
+        It does not return anything but directly writes the Commandbyte and new Divisor to the device.
+        """
+        if self.__isModernDevice():
+            self.__Ft232.write(bytes([0x8a]))
+
+    def __enableThreePhaseClocking(self):
+        """ Missing description.
+        This function implements chapter 6.3 Enable 3 Phase Data Clocking.
+        It does not return anything but directly writes the Commandbyte and new Divisor to the device.
+        """
+        if self.__isModernDevice():
+            self.__Ft232.write(bytes([0x8c]))
+
+    def __disableThreePhaseClocking(self):
+        """ Missing description.
```

Comment:
Why do all these functions say "Missing Description"? Probably not useful and should be removed.

Reply:
Those are functions which are part of the MPSSE "API" out of the  https://www.ftdichip.com/Support/Documents/AppNotes/AN_108_Command_Processor_for_MPSSE_and_MCU_Host_Bus_Emulation_Modes.pdf which i dont really understand and do not need for SPI. But they 2 could be important or interisting for other custom protocols. Maybe "Not implemented" schould be changed to "not fully implemended yet", so that someone who needs it can easily implemend it.

Reply:
Uh, yeah if the code is not necessary for anything, no one needs it right now, you didn't test it, and you only added it because it's in the documentation: then leave it out completely. Let someone who needs it, and can test it, send me a PR to add it. 

I don't want to maintain code that nobody wants and might not even be correct.